components/create_comment/create_comment.jsx

```diff
@@ -465,17 +490,45 @@ class CreateComment extends React.PureComponent {
             this.setState({draft: updatedDraft});
         }
 
-        if (enableConfirmNotificationsToChannel && useChannelMentions &&
+        const notificationsToChannel = enableConfirmNotificationsToChannel && useChannelMentions;
+        let memberNotifyCount = 0;
+        let channelTimezoneCount = 0;
+        let mentions = [];
+        const notContainsAtChannel = !containsAtChannel(draft.message);
+        if (enableConfirmNotificationsToChannel && notContainsAtChannel) {
+            // Groups mentioned in users text
+            mentions = groupsMentionedInText(draft.message, allowReferencedGroups);
+            if (mentions.length > 0) {
+                mentions = mentions.
+                    map((group) => {
+                        const mappedValue = channelMemberCountsByGroup[group.id];
+                        if (mappedValue && mappedValue.channel_member_count > Constants.NOTIFY_ALL_MEMBERS && mappedValue.channel_member_count > memberNotifyCount) {
+                            memberNotifyCount = mappedValue.channel_member_count;
+                            channelTimezoneCount = mappedValue.channel_member_timezones_count;
+                        }
+                        return `@${group.name}`;
+                    });
+            }
+        }
+
+        if (notificationsToChannel &&
             channelMembersCount > Constants.NOTIFY_ALL_MEMBERS &&
-            containsAtChannel(this.state.draft.message)) {
+            notContainsAtChannel) {
```

Comment:
why is this `notContainsAtChannel`?

Reply:
True that! Done! It had to be `!notContainsAtChannel`